src/flb_log.c

```diff
@@ -507,6 +603,16 @@ struct flb_log *flb_log_create(struct flb_config *config, int type,
         return NULL;
     }
 
+    /* Create metrics */
+    log->metrics = flb_log_metrics_create();
+    if (log->metrics == NULL) {
+        fprintf(stderr, "[log] could not create log metrics\n");
+        mk_event_loop_destroy(log->evl);
+        flb_free(log);
+        config->log = NULL;
```

Comment:
While looking at this I noticed that `config->log` is set immediately after allocating it. 

I've checked the client code and verified that there are no threading issues, however, I wanted to use this as an example of bad logic for the new contributors.

The problem is that the uninitialized log context is set to `config` which means it could be prematurely accessed by other threads. The right thing would be to only set `config->log` at the end of this function when we know for sure that the context is fully initialized and usable.

You don't need to change anything here, just look at it and if you have any questions ask them.

Reply:
Makes sense to me.